### ISSUE TITLE
Add field for cleaner information

### ISSUE THREAD

**Issue by JamesBremner** [JamesBremner/Agents2Tasks#26]
>  we probably need another field to provide the cleaner information(A or B or C etc) separately to family field.

( This was buried in what looked like a bug report #25 )

**Comment by JamesBremner** [comment]
What do you want the algorithm to do with the new cleaner field?

**Comment by rejkid** [comment]
I am wording it. Just give me a minute please

**Comment by rejkid** [comment]
Can we make an enhancement so that the task "Cleaner" is treated like a separate entity - it is still a task by nature but belongs to a specific group (A, B or C etc). I would have a separate field.

Algorithm for assigning tasks Cleaner to time slots:
 
If the agent can do "Cleaner" task, he/she will belong to one group (A or B or C etc).
A group is a set of agents working together on a particular task; in this case the "Cleaner" task. 
Groups perform their duties on a weekly basis (as per time slot assignments). For example, group "A" does cleaning on one Saturday, then group B does the following Saturday. This continues with the groups until all groups have done their duties once and then the process of assigning starts from the beginning.
When assigning Cleaner agents to time slots, the algorithm should assign as many agents (starting from group A)  as is specified by the the time slot. For the next time slot it should assign as many agents from Group B as specified by the time slot etc, until  all groups are used . It should then start assigning Cleaner agents to time slots again starting from group A.

**Comment by JamesBremner** [comment]
I will think about it over the weekend.

In the meantime could you, as time allows:

1 Edit this statement "Algorithm for assigning tasks Cleaner to time slots:" which makes no sense.  Tasks are specified for timeslots.  Agents are assigned to tasks  

2. Take a look at the three issues that are assigned to you in test.  There is little point in adding enhancements when previous enhancement and bugs remain untested.

![image](https://github.com/JamesBremner/Agents2Tasks/assets/2046227/74fd55f7-5bac-4a85-806a-b04ad5dae895)

**Comment by rejkid** [comment]
Agreed

**Comment by JamesBremner** [comment]
> group "A" does cleaning on one Saturday, then group B does the following Saturday. 

The current algorithm handles this quite nicely:

```
a ismith@gmail.com&1961-07-02 1 A Cleaner
a rejkid@gmail.com&1961-03-24 1 A Cleaner
a asmith@gmail.com&1998-06-07 1 A Cleaner 
a ablack@gmail.com&1998-06-07 1 B Cleaner 
a awhite@gmail.com&1998-06-07 1 B Cleaner 
a iblack@gmail.com&1961-07-02 1 B Cleaner 
a iwhite@gmail.com&1961-07-02 1 B Cleaner 
a mwalsh@gmail.com&2010-01-01 1 mwalsh@gmail.com MAS 
a osiekkolawy@gmail.com&1961-03-24 1 osiekkolawy@gmail.com MAS 
a jsmith@gmail.com&1961-03-24 1 jsmith@gmail.com EMHC Reader2 
a rejkid@gmail.com&1961-07-02 1 rejkid@gmail.com Reader1 
a jblack@gmail.com&1961-03-24 1 jblack@gmail.com EMHC Reader1 Acolyte 
a rejkid@gmail.com&1998-06-07 1 rejkid@gmail.com Reader2 
a rejkid@gmail.com&1992-08-28 1 rejkid@gmail.com MAS 
a jwhite@gmail.com&1961-03-24 1 jwhite@gmail.com EMHC Reader1 
A 20231118000 ismith@gmail.com&1961-07-02 to Cleaner
A 20231118000 asmith@gmail.com&1998-06-07 to Cleaner
A 20231118000 rejkid@gmail.com&1961-03-24 to Cleaner
A 20231125000 ablack@gmail.com&1998-06-07 to Cleaner
A 20231125000 iwhite@gmail.com&1961-07-02 to Cleaner
A 20231125000 awhite@gmail.com&1998-06-07 to Cleaner
```

Here is the input file

**Comment by rejkid** [comment]
Your date time has only 11 chars

**Comment by JamesBremner** [comment]
Fixed

[ex26.txt](https://github.com/JamesBremner/Agents2Tasks/files/13398347/ex26.txt)

**Comment by JamesBremner** [comment]
I think group agents are a better solution for the reasons described in #27

**Comment by JamesBremner** [comment]
I don't know why this is assigned to me.  I think it should be closed, because you have agreed to go with #27.
